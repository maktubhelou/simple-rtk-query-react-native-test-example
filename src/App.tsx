/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * Generated with the TypeScript template
 * https://github.com/react-native-community/react-native-template-typescript
 *
 * @format
 */

import React from 'react';
import _ from 'lodash';
import {
  Image,
  SafeAreaView,
  ScrollView,
  StatusBar,
  StyleSheet,
  Text,
  useColorScheme,
  View,
} from 'react-native';

import { Provider } from 'react-redux';

import { Colors } from 'react-native/Libraries/NewAppScreen';
import { store } from './store';
import { useGetPokemonByNameQuery } from './services/pokemon';

const Section: React.FC<{
  title: string;
}> = ({ children, title }) => {
  const isDarkMode = useColorScheme() === 'dark';
  return (
    <View style={styles.sectionContainer}>
      <Text
        style={[
          styles.sectionTitle,
          {
            color: isDarkMode ? Colors.white : Colors.black,
          },
        ]}>
        {title}
      </Text>
      <Text
        style={[
          styles.sectionDescription,
          {
            color: isDarkMode ? Colors.light : Colors.dark,
          },
        ]}>
        {children}
      </Text>
    </View>
  );
};

const Home = () => {
  const isDarkMode = useColorScheme() === 'dark';
  const { data, error, isLoading } = useGetPokemonByNameQuery('rattata');

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
  };

  return (
    <Provider store={store}>
      <SafeAreaView style={backgroundStyle}>
        <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
        <ScrollView
          contentInsetAdjustmentBehavior="automatic"
          style={backgroundStyle}>
          <View
            style={{
              backgroundColor: isDarkMode ? Colors.black : Colors.white,
            }}>
            <Section title="Pokemon">
              Select your favorite{' '}
              <Text style={styles.highlight}>Pokemon </Text>
              from the list below.
            </Section>
            <Section title="Example">
              {error ? (
                <Text>Oh no, there was an error!</Text>
              ) : isLoading ? (
                <Text>Loading...</Text>
              ) : data ? (
                <View>
                  <Text>{_.capitalize(data.species.name)}</Text>
                  <Image
                    source={{ uri: data.sprites.front_shiny }}
                    style={styles.thumbnail}
                  />
                </View>
              ) : null}
            </Section>
          </View>
        </ScrollView>
      </SafeAreaView>
    </Provider>
  );
};

const App = () => {
  return (
    <Provider store={store}>
      <Home />
    </Provider>
  );
};

const styles = StyleSheet.create({
  sectionContainer: {
    marginTop: 32,
    paddingHorizontal: 24,
  },
  thumbnail: {
    width: 100,
    height: 100,
  },
  sectionTitle: {
    fontSize: 24,
    fontWeight: '600',
  },
  sectionDescription: {
    marginTop: 8,
    fontSize: 18,
    fontWeight: '400',
  },
  highlight: {
    fontWeight: '700',
  },
});

export default App;
