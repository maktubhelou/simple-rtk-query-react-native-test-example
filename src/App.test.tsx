/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from './App';
import { renderWithProviders } from '../test-utils';

describe('App', () => {
  it('matches snapshot', async () => {
    const component = <App />;
    const { store, toJSON } = renderWithProviders(component);

    expect(toJSON()).toMatchSnapshot();
    expect(store?.getState()).toMatchSnapshot();
  });
});
