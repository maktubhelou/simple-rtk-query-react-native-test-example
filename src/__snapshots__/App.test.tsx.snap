// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#F3F3F3",
    }
  }
>
  <RCTScrollView
    contentInsetAdjustmentBehavior="automatic"
    style={
      Object {
        "backgroundColor": "#F3F3F3",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#FFF",
          }
        }
      >
        <View
          style={
            Object {
              "marginTop": 32,
              "paddingHorizontal": 24,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 24,
                  "fontWeight": "600",
                },
                Object {
                  "color": "#000",
                },
              ]
            }
          >
            Pokemon
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 8,
                },
                Object {
                  "color": "#444",
                },
              ]
            }
          >
            Select your favorite 
            <Text
              style={
                Object {
                  "fontWeight": "700",
                }
              }
            >
              Pokemon 
            </Text>
            from the list below.
          </Text>
        </View>
        <View
          style={
            Object {
              "marginTop": 32,
              "paddingHorizontal": 24,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 24,
                  "fontWeight": "600",
                },
                Object {
                  "color": "#000",
                },
              ]
            }
          >
            Example
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 8,
                },
                Object {
                  "color": "#444",
                },
              ]
            }
          >
            <Text>
              Loading...
            </Text>
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`App matches snapshot 2`] = `
Object {
  "pokemonApi": Object {
    "config": Object {
      "focused": true,
      "keepUnusedDataFor": 0,
      "middlewareRegistered": false,
      "online": true,
      "reducerPath": "pokemonApi",
      "refetchOnFocus": false,
      "refetchOnMountOrArgChange": false,
      "refetchOnReconnect": false,
    },
    "mutations": Object {},
    "provided": Object {},
    "queries": Object {},
    "subscriptions": Object {},
  },
}
`;
